mne/io/reference.py

```diff
@@ -522,11 +521,11 @@ def set_bipolar_reference(inst, anode, cathode, ch_name=None, ch_info=None,
         inst.info['chs'][an_idx]['ch_name'] = name
         logger.info('Bipolar channel added as "%s".' % name)
         inst.info._update_redundant()
-        if an in rem_ca:
-            idx = rem_ca.index(an)
-            del rem_ca[idx]
 
-    # Drop remaining cathode channels
-    inst.drop_channels(rem_ca)
+    # Drop remaining channels.
+    inst.drop_channels([
+        ch_name for ch_name in set(list(anode) + list(cathode))
```

Comment:
would the somewhat more concise `for ch_name in set(anode) + set(cathode)` also work?

Reply:
... or, couldn't you just do a set union? `drop_channels = (set(anode) + set(cathode)).union(set(inst.ch_names))` or so

Reply:
Using only set operations, we could do the following:
```
drop_channels = (set(anode) | set(cathode)) & set(inst.ch_names)
inst.drop_channels(drop_channels)
``` 
I like this option :)